tests/test_quickfiles.py

```diff
@@ -0,0 +1,60 @@
+# -*- coding: utf-8 -*-
+import pytest
+
+from osf.models import QuickFiles, BaseFileNode
+from osf_tests.factories import AuthUserFactory
+
+from tests.base import test_app
+from webtest_plus import TestApp
+from addons.osfstorage.tests.utils import make_payload
+
+from framework.auth import signing
+
+
+@pytest.fixture()
+def user():
+    return AuthUserFactory()
+
+
+@pytest.fixture()
+def quickfiles(user):
+    return QuickFiles.objects.get(creator=user)
+
+
+@pytest.fixture()
+def flask_app():
+    return TestApp(test_app)
+
+
+@pytest.fixture()
+def post_to_quickfiles(quickfiles, user, flask_app, **kwargs):
+    def func(name, *args, **kwargs):
+        osfstorage = quickfiles.get_addon('osfstorage')
+        root = osfstorage.get_root()
+        url = '/api/v1/project/{}/osfstorage/{}/children/'.format(quickfiles._id, root._id)
+        expect_errors = kwargs.pop('expect_errors', False)
+        payload = make_payload(user=user, name=name, **kwargs)
+
+        res = flask_app.post_json(url, signing.sign_data(signing.default_signer, payload), expect_errors=expect_errors)
+        return res
+
+    return func
+
+
+@pytest.mark.django_db
+class TestUserQuickFilesFileCreation:
+
+    def test_create_file(self, quickfiles, user, post_to_quickfiles):
+        name = 'WoopThereItIs.pdf'
+
+        res = post_to_quickfiles(name)
+
+        assert res.status_code == 201
+        assert res.json['status'] == 'success'
+        assert BaseFileNode.objects.filter(type='osf.osfstoragefile', node__creator=user, name=name).exists()
+
+    def test_create_folder_throws_error(self, flask_app, user, quickfiles, post_to_quickfiles):
+        name = 'new_illegal_folder'
+        res = post_to_quickfiles(name, kind='folder', expect_errors=True)
+
+        assert res.status_code == 403
```

Comment:
403 is used specifically for authorization/permissions.

RFC7231 has more complete definitions:

> If authentication credentials were provided in the request, the server considers them insufficient to grant access. 

https://tools.ietf.org/html/rfc7231#section-6.5.3

In this case, the client is sending a semantically invalid request--not an unauthorized request--and therefore a 400 is appropriate.